src/middlewared/middlewared/plugins/nfs.py

```diff
@@ -454,13 +457,17 @@ async def resolve(hostname):
             if domain := get_wildcard_domain(hostname):
                 hostname = domain
 
-            try:
-                return (
-                    await asyncio.wait_for(self.middleware.run_in_thread(socket.getaddrinfo, hostname, None), 5)
-                )[0][4][0]
-            except Exception as e:
-                self.logger.warning("Unable to resolve host %r: %r", hostname, e)
+            if leftmost_has_wildcards(hostname):
+                # We know this will not resolve
                 return None
+            else:
```

Comment:
Since you're here and making changes, we have a dns client that is async native so no reason to create a native thread to run `socket.getaddrinfo`. Name of the function is `dnsclient.forward_lookup` which takes a list of hostnames. You should be able to grep our code-base and see example usages of this.